### PR TITLE
Update cache on netlink error

### PR DESCRIPTION
**Pull request by ChitrangSrivastava**
## Prerequisites

- [x] I have read the [Contributing Guidelines](../blob/main/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](../blob/main/CODE_OF_CONDUCT.md).
- [x] If this is a non-trivial change, I have already opened an accompanying Issue.
- [x] If applicable, I have included documentation updates alongside my code changes.

## Description

fib_linux was throwing netlink error when e2e_minion is restarted.
```
I0904 17:23:13.605388 24895 NetlinkFibHandler.cpp:105] Adding/Updating unicast routes of client OPENR, numRoutes=5
E0904 17:23:13.609419 24882 NetlinkProtocolSocket.cpp:171] Netlink request error for seq=350, retval=-19
E0904 17:23:13.610333 24882 NetlinkProtocolSocket.cpp:171] Netlink request error for seq=352, retval=-19
E0904 17:23:13.611045 24882 NetlinkProtocolSocket.cpp:171] Netlink request error for seq=354, retval=-19
E0904 17:23:13.611716 24882 NetlinkProtocolSocket.cpp:171] Netlink request error for seq=356, retval=-19
E0904 17:23:13.612370 24882 NetlinkProtocolSocket.cpp:171] Netlink request error for seq=358, retval=-19
```
#### Root Cause
When openr recevied a prefix on any interface , it ask fib to program a route for this.
Here fib_linux uses netlink to program route in Linux.
platform_linux convert interface name to interface index, this is done through a cache which is initialized at startup.
Now when e2e_minion is restarted terra interfaces are recreated and hence there interface index are changed and platform_linux didnt updated its cache to send netlink mesage to updated index.
Thats the reason we see error in netlink.

#### Fix
if netlink reports error , set a flag to reload interface index cache on next requtest
logs after fix
```
I0913 02:07:51.510177 30302 NetlinkFibHandler.cpp:128] Deleting unicast routes of client OPENR, numRoutes=5
I0913 02:08:16.183758 30302 NetlinkFibHandler.cpp:108] Adding/Updating unicast routes of client OPENR, numRoutes=5
I0913 02:08:16.184090 30302 NetlinkFibHandler.cpp:555] ifIndex 101 is invalid, setting flag to reload ifIndex cache on next request
E0913 02:08:16.188894 30288 NetlinkProtocolSocket.cpp:171] Netlink request error for seq=15, retval=-19
E0913 02:08:16.253119 30305 service_tcc.h:132] openr::fbnl::NlException: Error(19) - No such device. One or more netlink request failed  in function addUnicas
I0913 02:08:16.261253 30305 Cpp2Connection.cpp:283] ERROR: Task killed: Rocket upgrade disabled: ::1
I0913 02:08:16.264134 30302 NetlinkFibHandler.cpp:108] Adding/Updating unicast routes of client OPENR, numRoutes=5
```

## Test Plan

1. restart e2e_minion
2. reboot node
3. bring the link down
4. reboot POP node

### CODE REVIEW

**Review by MatthewWEdwards — APPROVED** [review]

**Comment by MatthewWEdwards** [comment]
Waiting for Cambium to sign the CLA before merging

File: recipes-support/openr/files/0001-fix-openr-update-interface-index-cache-on-netlink-er.patch
Comment (ChitrangSrivastava) [reply recipes-support/openr/files/0001-fix-openr-update-interface-index-cache-on-netlink-er.patch]: Fixed

File: recipes-support/openr/files/0001-fix-openr-update-interface-index-cache-on-netlink-er.patch
Comment (ChitrangSrivastava) [reply recipes-support/openr/files/0001-fix-openr-update-interface-index-cache-on-netlink-er.patch]: Fixed

**Pull request merged**
